tests/main/enable-disable-units-gpio/task.yaml

```diff
@@ -0,0 +1,88 @@
+summary: Check that systemd units are enabled/disabled and gpio works after rebooting
+
+details: |
+    This test makes sure that the systemd snippet created by the gpio interface
+    is executed after a reboot.
+
+    It modifies the core snap to provide a gpio slot. Also, a mocked gpio node and the
+    required systemfs files (export and unexeport) are created as a bind mount. The test
+    expects that, after a snap declared a gpio plug is installed and connected, after
+    a reboot the systemd service tries to regenerate the gpio device node if it does not
+    find it.
+
+systems: [ubuntu-core-16-64]
+
+environment:
+    GPIO_MOCK_DIR: /home/test/gpio-mock
+
+prepare: |
+    if [ "$TRUST_TEST_KEYS" = "false" ]; then
+        echo "This test needs test keys to be trusted"
+        exit
+    fi
+    echo "Given a snap declaring a plug on gpio is installed"
+    . $TESTSLIB/snaps.sh
+    install_local gpio-consumer
+
+    echo "And a mocked gpio device is in place"
+    cat > /home/test/gpio-mock.sh <<-EOF
+    #!/bin/sh
+    if [ ! -d "$GPIO_MOCK_DIR" ]; then
+        # the service has just been created
+        mkdir -p $GPIO_MOCK_DIR
+        touch $GPIO_MOCK_DIR/gpio100 $GPIO_MOCK_DIR/export $GPIO_MOCK_DIR/unexport
+    else
+        # after reboot, remove device node to test export
+        rm $GPIO_MOCK_DIR/gpio100
+        :> $GPIO_MOCK_DIR/export
```

Comment:
While I know what this does, it is a bit mysterious. Why don't we just `truncate` this file?